coral-service/src/main/java/com/linkedin/coral/coralservice/controller/VisualizationController.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.coralservice.controller;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.linkedin.coral.coralservice.entity.VisualizationRequestBody;
+import com.linkedin.coral.coralservice.entity.VisualizationResponseBody;
+import com.linkedin.coral.coralservice.utils.RewriteType;
+
+import static com.linkedin.coral.coralservice.utils.VisualizationUtils.*;
+
+
+@RestController
+@RequestMapping("/api/visualizations")
+public class VisualizationController {
+  private File imageDir = createImageDir();
+
+  @PostMapping("/generategraphs")
+  public ResponseEntity getIRVisualizations(@RequestBody VisualizationRequestBody visualizationRequestBody) {
+    final String fromLanguage = visualizationRequestBody.getFromLanguage();
+    final String query = visualizationRequestBody.getQuery();
+    final RewriteType rewriteType = visualizationRequestBody.getRewriteType();
+
+    UUID sqlNodeImageID, relNodeImageID;
+    UUID postRewriteSqlNodeImageID = null;
+    UUID postRewriteRelNodeImageID = null;
```

Comment:
Not clear why some are explicitly initialized with null and some are not. Let us also defined each variable in its owns statement.

Reply:
Java throws ```Variable 'postRewriteSqlNodeImageID' might not have been initialized ``` for `postRewriteSqlNodeImageID` and `postRewriteRelNodeImageID` since they're nested deeper.

I will initialize all 4 variables the same way for consistency.